### PR TITLE
refactor(FOROME-786): change logics of slice page items & select all btn

### PR DESCRIPTION
**Pull request by ghost**
PR into refactor/FOROME-744 branch

https://quantori.atlassian.net/browse/FOROME-786

### CODE REVIEW

File: src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx [hunk 1]
```diff
@@ -42,7 +41,12 @@ export const ModalFilters = observer((): ReactElement => {
         .includes(modalFiltersStore.searchValue.toLocaleLowerCase()),
   )
 
-  const chunks = createChunks(filteredGroupList, groupsPerPage)
+  const groupsPage = filteredGroupList.slice(
+    currentPage * groupsPerPage,
+    (currentPage + 1) * groupsPerPage,
+  )
+
+  const pagesNumbers = Math.ceil(filteredGroupList.length / groupsPerPage)
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx]: you can move the whole bunch of logic into store getters

Comment (ghost) [reply src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx]: Moved

**Review by Lapk1n — APPROVED** [review]

File: src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx (hunk 1 above)
Comment (AndreyUstyumenko) [reply src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx]: rename modal-filters.store.tsx to modal-filters.store.ts

Comment (ghost) [reply src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx]: Renamed

**Review by AndreyUstyumenko — APPROVED** [review]

**Review by AlMaQntr — APPROVED** [review]

**Pull request merged**